### PR TITLE
8339975: Open some dialog awt tests 2

### PR DESCRIPTION
**Pull request by alisenchung**
Second set of dialog swing tests to open during fall 2024 test sprint

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339975](https://bugs.openjdk.org/browse/JDK-8339975): Open some dialog awt tests 2 (**Bug** - P4)

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21488/head:pull/21488` \
`$ git checkout pull/21488`

Update a local copy of the PR: \
`$ git checkout pull/21488` \
`$ git pull https://git.openjdk.org/jdk.git pull/21488/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21488`

View PR using the GUI difftool: \
`$ git pr show -t 21488`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21488.diff">https://git.openjdk.org/jdk/pull/21488.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21488#issuecomment-2410306607)

### CODE REVIEW

File: test/jdk/java/awt/Dialog/DialogDisposeLeak.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * @summary Test for bug(s): 4193022, disposing dialog leaks memory
+ * @key headful
+ * @run main/othervm -Xmx128m DialogDisposeLeak
+ */
+
+public class DialogDisposeLeak {
+    static Frame frame;
+
+    public static void main(String args[]) throws Exception {
+        EventQueue.invokeLater(() -> {
```
Comment (azvegint) [inline test/jdk/java/awt/Dialog/DialogDisposeLeak.java:42]:
```suggestion
        EventQueue.invokeAndWait(() -> {
```

The test may prematurely end with `invokeLater` even before it starts testing anything, because jtreg will kill it after exiting the main.

File: test/jdk/java/awt/Dialog/DialogDisposeLeak.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+/*
+ * @test
+ * @bug 4193022
+ * @summary Test for bug(s): 4193022, disposing dialog leaks memory
+ * @key headful
+ * @run main/othervm -Xmx128m DialogDisposeLeak
+ */
+
+public class DialogDisposeLeak {
```
Comment (azvegint) [inline test/jdk/java/awt/Dialog/DialogDisposeLeak.java:38]: It looks like this test was oversimplified, as I no longer see steps to reproduce the original issue.

https://bugs.openjdk.org/browse/JDK-4193022

> Container.trackMouseEnterExit() calls startListeningForOtherDrags() when the mouse pointer enters the container. It calls stopListeninForOtherDrags() when the mouse exits the container.
>
> If the container is a window (in our case a JDialog), and the dispose() method is called while the mouse pointer is within the window (user clicks on the 'ok' button, for example), then the window is disposed, but no mouse exit event occurs.
>
> This causes the window and everything it points to not to be collectable, as there is still the Container drag listener pointing at the window.

File: test/jdk/java/awt/TextArea/TextAreaHScrollbarTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/*
```
Comment (azvegint) [inline test/jdk/java/awt/TextArea/TextAreaHScrollbarTest.java:1]: These text area tests appear to be unrelated.

File: test/jdk/java/awt/Dialog/FileDialogTest.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+import java.awt.Panel;
+import java.awt.TextField;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+/*
+ * @test
+ * @bug 4105025 4153487 4177107 4146229 4119383 4181310 4152317
+ * @summary Test: FileDialogTest
+ * @library /open/test/jdk/java/awt/regtesthelpers
```
Comment (azvegint) [inline test/jdk/java/awt/Dialog/FileDialogTest.java:39]: This won't compile, please fix the path here and in the other tests

```suggestion
 * @library /java/awt/regtesthelpers
```

File: test/jdk/java/awt/Dialog/DialogDisposeLeak.java (hunk 1 above)
Comment (alisenchung) [reply test/jdk/java/awt/Dialog/DialogDisposeLeak.java:42]: the problem with using invokeAndWait is that the test uses a modal dialog and so everything is blocked once the dialog is shown in an invokeAndWait

Comment (azvegint) [reply test/jdk/java/awt/Dialog/DialogDisposeLeak.java:42]: Try to run it with jtreg and `invokeLater`, you will not see the appearing/disappearing dialog, so it doesn't have time to check anything.
This is almost the same as running a jtreg test with the empty main method, so the test appears to be useless as it is now.

> the problem with using invokeAndWait is that the test uses a modal dialog and so everything is blocked once the dialog is shown in an invokeAndWait

It seems that you need to rewrite the test properly, using the robot off the EDT to interact with a mouse (as it was in the original test, because it is critical according to [JDK-4193022](https://bugs.openjdk.org/browse/JDK-4193022)), and all the stuff we usually use for the test stability(`robot.waitForIdle(); robot.delay(500)` etc.)

File: test/jdk/java/awt/Dialog/DialogDisposeLeak.java (hunk 2 above)
Comment (alisenchung) [reply test/jdk/java/awt/Dialog/DialogDisposeLeak.java:49]: i will revert this test to a manual test

**Comment by alisenchung** [comment]
Instead of rewriting the test I've instead just reverted DialogDisposeLeak to a manual test

File: test/jdk/java/awt/Dialog/DialogDisposeLeak.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    public void testOK() {
+        label.setText("Test has passed. Dialog finalized.");
+    }
+}
+
+class DisposeDialog extends Dialog {
+    DisposeDialog(Frame frame) {
+        super(frame, "DisposeDialog", true);
```
Comment (azvegint) [inline test/jdk/java/awt/Dialog/DialogDisposeLeak.java:94]: The dialog to click on is too far away from the parent frame, it will be easier to click 10 times if it is just under the mouse cursor.

```suggestion
        super(frame, "DisposeDialog", true);
        setLocation(frame.getX(), frame.getY());
```

**Review by azvegint — APPROVED** [review]

**Comment by alisenchung** [comment]
@azvegint can you reapprove? just added the changes to move dialog closer

**Review by azvegint — APPROVED** [review]
